CMakeLists.txt

```diff
@@ -122,6 +123,7 @@ if("virtual" IN_LIST COMPILE_TARGETS)
   )
 
   target_compile_options(ccf.virtual PUBLIC -stdlib=libc++)
+  add_warning_checks(ccf.virtual)
```

Comment:
As discussed, it would have been nicer to have this be applied automatically by directory than something we need to manually opt into, but I guess it wasn't possible.

Reply:
Agreed. I think this bigger cmake refactoring should be done in a different PR though.